### ISSUE TITLE
Vulnerability detector refactor failing check analysis

### ISSUE THREAD

**Issue by MiguelazoDS** [wazuh/wazuh#22144]
# Description 

It was found as part of this issue https://github.com/wazuh/wazuh/issues/21718 that some checks are not passing due to a JSON exception. 

https://github.com/wazuh/wazuh/pull/22013/checks?check_run_id=21979395255

```console
2024/02/26 11:44:42 wazuh-modulesd:vulnerability-scanner: WARNING: Exception during re-scan: basic_string::_M_construct null not valid
2024/02/26 11:45:02 wazuh-modulesd:content-updater: INFO: Data published
2024/02/26 11:45:02 wazuh-modulesd:vulnerability-scanner: INFO: Processing message
2024/02/26 11:45:02 wazuh-modulesd:content-updater: INFO: Action for 'vulnerability_feed_manager' finished
2024/02/26 11:45:02 wazuh-modulesd:vulnerability-scanner: INFO: Processing file: queue/vd_updater/tmp/contents/240277-api_file.json
2024/02/26 11:45:03 wazuh-modulesd:vulnerability-scanner: ERROR: Error processing message: [json.exception.out_of_range.401] array index 195 is out of range
2024/02/26 11:45:03 wazuh-modulesd:content-updater: INFO: Starting on-demand action for 'vulnerability_feed_manager'
2024/02/26 11:45:03 wazuh-modulesd:content-updater: INFO: Action for 'vulnerability_feed_manager' started
2024/02/26 11:45:21 indexer-connector: WARNING: Error initializing IndexerConnector: No available server, we will try again after 60 seconds.
```

- [ ] Find the root cause of the issue.

**Comment by Dwordcito** [comment]
@MiguelazoDS we need the CI/CD execution! to reproduce this issue.

**Comment by MiguelazoDS** [comment]
Already fixed here. 

https://github.com/wazuh/wazuh/commit/5944968608f924dde3fc65ddb8b92040d545a22a

**Comment by sebasfalcone** [comment]
## Resolution
Analysis concluded.


### Duplicated
- `WARNING: Exception during re-scan: basic_string::_M_construct null not valid`
  - Fixed by: #22068

### New issues
- `WARNING: Failed to scan package: '', CVE Numbering Authorities (CNA): 'nvd', Error: 'Invalid package/cna name.'` 
  - #22137
